### PR TITLE
feat: add mixed support to task params (MAPCO-4210)

### PR DESCRIPTION
**Pull request by asafMasa**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Add task support for Mixed format in GPKG file - "outputFormatStrategy": "mixed"

### CODE REVIEW

File: src/clients/jobManagerWrapper.ts [hunk 1]
```diff
@@ -81,6 +82,7 @@ export class JobManagerWrapper extends JobManagerClient {
           parameters: {
             isNewTarget: true,
             targetFormat: data.targetFormat,
+            outputFormatStrategy: TileFormatStrategy.Mixed,
```
Comment (shimoncohen) [inline src/clients/jobManagerWrapper.ts]: This is already passed as `mixed` through `data` from `createPackageManager`.
I think you should either use from data or don't pass with data.

File: src/clients/jobManagerWrapper.ts [hunk 2]
```diff
@@ -129,6 +131,7 @@ export class JobManagerWrapper extends JobManagerClient {
...
```
Comment (shimoncohen) [inline src/clients/jobManagerWrapper.ts]: Same here.

File: tests/mocks/data.ts
Comment (shimoncohen) [inline tests/mocks/data.ts:1]: It looks like you changed all tests to use `mixed`, what about some tests with `fixed`?

File: src/common/enums.ts [hunk 3]
```diff
@@ -4,4 +4,11 @@ enum ArtifactType {
   GPKG = 'GPKG',
 }
 
+export enum TileFormatStrategy {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  Fixed = 'fixed',
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  Mixed = 'mixed',
+}
```
Comment (shimoncohen) [inline src/common/enums.ts]: Why are you disabling `eslint` on these lines?

**Review by shimoncohen — CHANGES_REQUESTED** [review]

File: src/clients/jobManagerWrapper.ts (hunk 1 above)
Comment (asafMasa) [reply src/clients/jobManagerWrapper.ts]: fixed

File: src/clients/jobManagerWrapper.ts (hunk 2 above)
Comment (asafMasa) [reply src/clients/jobManagerWrapper.ts]: fixed

File: src/common/enums.ts (hunk 3 above)
Comment (asafMasa) [reply src/common/enums.ts]: Because lint expects CAPITAL leters in enum values

File: tests/mocks/data.ts
Comment (asafMasa) [reply tests/mocks/data.ts:1]: added

File: src/common/enums.ts (hunk 3 above)
Comment (shimoncohen) [reply src/common/enums.ts]: And what's the problem to make it CAPITAL?

File: tests/mocks/data.ts
Comment (shimoncohen) [reply tests/mocks/data.ts:1]: I see you added a specific test for `fixed`, but why didn't you add cases to this data file?

File: src/common/enums.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/enums.ts]: As i know, its should be CAPITAL , ex. TileFormatStrategy.FIXED

File: tests/mocks/data.ts
Comment (asafMasa) [reply tests/mocks/data.ts:1]: because the default is `mixed`, and its the main scenario

File: src/common/enums.ts (hunk 3 above)
Comment (asafMasa) [reply src/common/enums.ts]: fixed

File: tests/mocks/data.ts
Comment (asafMasa) [reply tests/mocks/data.ts:1]: mixed test added

File: tests/unit/clients/jobManagerClient.spec.ts [hunk 4]
```diff
@@ -191,6 +194,62 @@ describe('JobManagerClient', () => {
           expect(postFun).toHaveBeenCalledTimes(1);
           expect(response).toStrictEqual(expectedResponse);
         });
+
+        it('should create Export job successfully with MIXED tiles strategy', async () => {
```
Comment (shimoncohen) [inline tests/unit/clients/jobManagerClient.spec.ts]: Why not pass strategy to test instead of creating two separate tests?

Comment (asafMasa) [reply tests/unit/clients/jobManagerClient.spec.ts]: fixed

**Review by shimoncohen — APPROVED** [review]

**Pull request merged**